### PR TITLE
Add support for the PSL data format

### PR DESCRIPTION
**Pull request by gregvonkuster**
Add support for the PSL data format https://genome.ucsc.edu/FAQ/FAQformat#format2

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by gregvonkuster** [comment]
> > Attempt to fix lint problems
> 
> You should be able to fix this kind of issues with `make format` from the galaxy root directory.

Thanks for letting me know - it's been a while since I submitted a PR here.  :)

**Comment by gregvonkuster** [comment]
Is the failing selenium test relevant?
```
=========================== short test summary info ============================
[4281](https://github.com/galaxyproject/galaxy/actions/runs/3275237777/jobs/5389824797#step:11:4282)
FAILED lib/galaxy_test/selenium/test_uploads.py::TestUploads::test_upload_paired_list
[4282](https://github.com/galaxyproject/galaxy/actions/runs/3275237777/jobs/5389824797#step:11:4283)
========== 1 failed, 79 passed, 30579 warnings in 2802.77s (0:46:42) ===========
[4283](https://github.com/galaxyproject/galaxy/actions/runs/3275237777/jobs/5389824797#step:11:4284)
Testing complete. HTML report is in "./run_selenium_tests.html".
[4284](https://github.com/galaxyproject/galaxy/actions/runs/3275237777/jobs/5389824797#step:11:4285)
Error: Process completed with exit code 1.
```

**Comment by gregvonkuster** [comment]
Associated with https://github.com/galaxyproject/tools-iuc/pull/4932

File: lib/galaxy/datatypes/registry.py [hunk 1]
```diff
@@ -711,6 +711,7 @@ def set_default_values(self):
                 "customtrack": interval.CustomTrack(),
                 "csfasta": sequence.csFasta(),
                 "fasta": sequence.Fasta(),
+                "psl": tabular.Psl(),
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/registry.py]:
```suggestion
```

File: lib/galaxy/datatypes/registry.py [hunk 2]
```diff
@@ -759,6 +760,7 @@ def set_default_values(self):
                 "maf": "text/plain",
                 "memexml": "application/xml",
                 "pileup": "text/plain",
+                "psl": "text/plain",
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/registry.py]:
```suggestion
```
please set these in the datatypes_conf.xml.sample file

File: lib/galaxy/datatypes/registry.py [hunk 3]
```diff
@@ -803,6 +805,7 @@ def set_default_values(self):
                 interval.Gff(),
                 interval.Gff3(),
                 tabular.Pileup(),
+                tabular.Psl(),
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/registry.py]:
```suggestion
```

File: lib/galaxy/datatypes/tabular.py [hunk 4]
```diff
@@ -1779,3 +1779,169 @@ def set_meta(self, dataset, overwrite=True, skip=None, max_data_lines=7, **kwd):
...
+            "tEnd",
+            "blockCount",
+            "blockSizes",
+            "qStarts",
+            "tStarts",
+        ]
+
+    def display_peek(self, dataset):
+        """Returns formated html of peek"""
+        return self.make_html_table(dataset, column_names=self.column_names)
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/tabular.py]:
```suggestion
```
part of TabularData

File: lib/galaxy/datatypes/tabular.py [hunk 5]
```diff
@@ -1779,3 +1779,169 @@ def set_meta(self, dataset, overwrite=True, skip=None, max_data_lines=7, **kwd):
...
+            if not line:
+                break
+            if line:
+                if line.startswith("browser") or line.startswith("track"):
+                    # Skip track lines.
+                    continue
+                items = line.split("\t")
+                if len(items) != 21:
+                    return False
+                # matches
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/tabular.py]: Can you pop out the items that are not supposed to be non-negative integers, validate those, then just check the rest with
```python
if any(int(item) < 0 for item in items):
    raise Exception("Out of range")
```
?

Comment (gregvonkuster) [reply lib/galaxy/datatypes/tabular.py]: All of the above should be fixed - thanks @mvdbeek

**Comment by gregvonkuster** [comment]
I don't think this PR has caused the failed selenium test.

**Comment by gregvonkuster** [comment]
@mvdbeek Can this be approved / merged, or is there something I've missed?  Thanks!

**Comment by mvdbeek** [comment]
Thanks @gregvonkuster !

**Pull request merged**